Fix 'computed property already' defined on the lesson summary page in Coach

MisRob opened the pull request:
## Summary

Fixes `The computed property "PageNames" is already defined in data.` on the lesson summary page in Coach (present on both `/lessons` and `/lessonstemp`)

![Screenshot from 2025-02-27 20-15-20](https://github.com/user-attachments/assets/7175f4fd-968c-4ca2-a09c-78c179d12090)

This happens because 'LessonSummaryPage' has 'PageNames' in its 'data' while at the same time importing 'commonCoach' mixin that contains 'PageNames' as computed.

Also adds 'this.' before 'PageNames' so there's no need for the global import.

## Reviewer guidance

- Go to Coach -> Lessons -> Select a lesson
- On the lesson summary page, there should be no regressions or console errors

MisRob committed:
Fix computer property already defined warning

Fixes 'The computed property "PageNames" is already
defined in data.'on the lesson summary page in Coach.

This happens because 'LessonSummaryPage' has
'PageNames' in its 'data' while at the same time importing
'commonCoach' mixin that contains 'PageNames' as computed.

Also adds 'this.' before 'PageNames' so there's no need
for the global import.

radinamatic commented:
Error `The computed property "PageNames" is already defined in data` is no more! 🎉 
Tested in Chrome and Firefox on Ubuntu 20.

Chrome | FF
-- | --
<img width="961" alt="chrome2" src="https://github.com/user-attachments/assets/5fe8c58e-11a9-45b4-8a8b-3f5bd785494b" /> | <img width="961" alt="ff2" src="https://github.com/user-attachments/assets/b94c0a09-b9cf-4def-8a4c-dccdc912b9d2" />

radinamatic reviewed (approved):
LGTM! 💯 👍🏽 :shipit:

rtibbles commented on kolibri/plugins/coach/assets/src/views/lessons/LessonSummaryPage/index.vue:
```diff
@@ -94,7 +94,6 @@
   import CoachAppBarPage from '../../CoachAppBarPage';
   import ReportsControls from '../../common/ReportsControls';
   import { REPORTS_LESSON_TABS_ID, ReportsLessonTabs } from '../../../constants/tabsConstants';
-  import { PageNames } from '../../../constants';
```
I think I'd rather maintain the explicit import here and reference the imported object.

I know this is technically available by virtue of the commonCoachMixin - but I'd prefer our direction of travel to be moving away from implicit uses of that, rather than towards.

rtibbles commented on kolibri/plugins/coach/assets/src/views/lessons/LessonSummaryPage/index.vue:
```diff
@@ -145,7 +144,6 @@
       const workingResourcesBackup = [...(this.$store.state.lessonSummary.workingResources || [])];
 
       return {
-        PageNames,
```
I think the diff can just be reduced to deleting this and making sure we're not referencing `this.PageNames` anywhere in the component.

As I suspect that linting saying this was undefined might have driven the inclusion here initially.

MisRob commented:
Thanks @rtibbles, I agree with all that. It was the first thing I tried, however using the explicit global import only unfortunately won't do because of template reference like [this one](https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/coach/assets/src/views/lessons/LessonSummaryPage/index.vue#L9-L11). Template doesn't have access to `PageNames` if it's not in `data` or `computed`.

I can inspect the page more closely and refactor its template to allow for what you suggest, it will just take a while to get back to it (I did this during the bugbash). Taking the above into account, do you still consider it blocking, or would follow-up be fine in case we need to get this fix in soon?

rtibbles commented:
Seems like this would still be compatible with what I am suggesting? The template references would still be able to use the mixin's value, we just revert the updates for non-template references to using `this.`

MisRob commented:
@rtibbles Ah sorry, I wrote it under an assumption we'd want to use the same source for all those occurrences. Yes, we can use mixin in the template, and the global import in the script.

rtibbles commented:
Yeah - not trying to make a tonne of work here - just think we can minimize the diff and keep things explicit where possible.

MisRob commented:
My thinking was that it may be a bit confusing to figure out from where it comes from if we mix them, but as long as we rely on mixins, one way or another there will be some kind of tension like this, so no problem if you see value in it.

MisRob commented:
Okay

MisRob committed:
Use explicit import

Replaces some occurences of `PageNames` being imported
via the common coach mixin by the global import
to make those references more explicit.

MisRob commented:
Updated @rtibbles. @radinamatic if you'd like to run check on this one more time before we merge, would be appreciated.

radinamatic commented:
With the latest asset I'm still not seeing the original error in the console, but there's a new one (both Firefox and Chrome on Ubuntu):

`[ERROR: kolibri/plugins/coach/assets/src/composables/useCoreCoach.js] Failed to obtain page title. Ensure that this route's meta.titleParts are corrrectly configured.`

<img width="961" alt="2025-03-05_14-47-16" src="https://github.com/user-attachments/assets/8082f906-868b-4bed-8443-3480bf8454fd" />

MisRob commented:
Hm I didn't notice that myself. I am not sure if it's related to these changes, will check on it. Thanks @radinamatic!

rtibbles commented:
I have a feeling that these errors can occasionally occur because of unpredictable loading order - not to say we shouldn't fix it, but I think it's likely this is not caused by @MisRob's changes.

MisRob commented:
@radinamatic @rtibbles I am unable to reproduce it, and agree that the nature of changes in this PR is 99% not the cause.
